openapi/p4pa-debt-position.openapi.yaml

```diff
@@ -0,0 +1,273 @@
+info:
+  title: p4pa-debt-positions
+  description: Api and Models
+  version: 0.0.1
+servers:
+  - url: http://localhost:8080
+    description: Generated server url
+paths:
+  /api/debt-positions:
+    post:
+      tags:
+        - debt-position-controller
+      operationId: createDebtPosition
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/DebtPositionDTO"
+        required: true
+      responses:
+        "201":
+          description: Created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DebtPositionDTO'
+        "400":
+          description: Invalid request
+        "500":
+          description: Internal server error
+components:
+  schemas:
+    DebtPositionDTO:
+      type: object
+      properties:
+        debtPositionId:
+          type: integer
+          format: int64
+        iupdOrg:
+          type: string
+        iupdPagopa:
+          type: string
+        description:
+          type: string
+        status:
+          type: string
+        ingestionFlowFileId:
+          type: integer
+          format: int64
+        ingestionFlowFileLineNumber:
+          type: integer
+          format: int64
+        organizationId:
+          type: integer
+          format: int64
+        debtPositionTypeOrgId:
+          type: integer
+          format: int64
+        notificationDate:
+          type: string
+          format: date-time
+        validityDate:
+          type: string
+          format: date-time
+        flagIuvVolatile:
+          type: boolean
+        creationDate:
+          type: string
+          format: date-time
+        updateDate:
+          type: string
+          format: date-time
+        paymentOptions:
+          type: array
+          items:
+            $ref: "#/components/schemas/PaymentOptionDTO"
+    InstallmentDTO:
+      type: object
+      properties:
+        installmentId:
+          type: integer
+          format: int64
+        paymentOptionId:
+          $ref: "#/components/schemas/PaymentOptionDTO"
+        status:
+          type: string
+        iud:
+          type: string
+        iuv:
+          type: string
+        iur:
+          type: string
+        dueDate:
+          type: string
+          format: date-time
+        paymentTypeCode:
+          type: string
+        amountCents:
+          type: integer
+          format: int64
+        notificationFeeCents:
+          type: integer
+          format: int64
+        remittanceInformation:
+          type: string
+        humanFriendlyRemittanceInformation:
+          type: string
+        balance:
+          type: string
+        legacyPaymentMetadata:
+          type: string
+        debtor:
+          $ref: "#/components/schemas/PersonDTO"
+        transfers:
+          type: array
+          items:
+            $ref: "#/components/schemas/TransferDTO"
+        receipt:
```

Comment:
Probably we will not need the ReceiptDTO to be referenced on the DebtPosition model...
I think that the useCase which could involve the Receipt would handle it separately.
Let's just set the id value, not refencing the model (don't remove the DTO, it could be usefull later ;) )

Reply:
modified